src/content/en/tools/chrome-devtools/javascript/overview.md

```diff
@@ -0,0 +1,140 @@
+reproduce the bug. Eliminating unnecessary steps like this helps you find
+and fix your bugs faster.
+
+### Step 2: Set and trigger a breakpoint
+
+Once you can reproduce the bug, it's time to start debugging with
+DevTools by setting a breakpoint.
+
+**Breakpoints** are intentional stopping places in your code.
+When the browser encounters a breakpoint, it pauses the script on
+that exact line of code.
+
+You set breakpoints in DevTools. DevTools provides many different types
+of breakpoints, but they all fall into one of two general categories:
+
+* **Line-of-code breakpoints**. Set on a specific line of code. Useful
+  when you want to inspect a particular section of code.
+* **Event-based breakpoints**. Set to trigger when a certain type of general
+  event occurs, like a `click` event or an uncaught exception. When the
+  event occurs, DevTools automatically takes you to the code associated
+  with the event. Event-based breakpoints are useful when you don't know
+  what code is causing a bug. They are often much faster to set than
+  line of code breakpoints, too.
+
+Ideally, you want to set your breakpoint somewhere close to the code that's
+causing the bug, so that you don't need to step through a bunch of code
+that's not related to your bug. The breakpoint needs to be set on a line of
+code before the bug, so that you
+can inspect what's happening in your code leading up to the bug. However,
+since you don't know what code is causing the bug, setting breakpoints is
+somewhat of a guessing game. It gets easier with practice.
+
+Once you've got your breakpoint set, interact with your page by reproducing
+the bug. This should trigger your breakpoint and pause the script.
+
+### Step 3: Step through the code
+
+Once your script is paused, use DevTools to step through it, one line of code
+at a time. Make sure that values are being set and passed as
+expected, and check that your script is executing in the expected order.
+
+### Step 4: Apply a fix
+
+Eventually, you'll find that a variable or function is producing a wrong
+value, or your script is executing in an unexpected order. You can apply
+a fix to your script directly within the DevTools UI, and then re-run your
+page to see if your change fixed the bug. If it did, remember to commit
+your change to your source code repository!
+
+### Step 5: Repeat, if necessary
+
+Debugging is a trial and error process. You might need to set breakpoints
+and walk through your code multiple times before you find a bug. Or, you
+might need to different possible reasons for why you're page is failing,
+and then test out each hypothesis individually.
+
+If you've tried different approaches and you still can't find the bug,
+sometimes it helps to reason out loud to yourself about what you expect
+your app to do as opposed to what it's actually
+doing. Seasoned programmers call this [rubber duck
+debugging](https://en.wikipedia.org/wiki/Rubber_duck_debugging). Other times,
```

Comment:
Seasoned programmers? I'm not sure about that. I never heard the term until Meggin came back from Grace Hopper.